imageioAWT integration test fails with latest 22.3.0-dev (libmanagement_ext.a no longer there)

As of late it's failing with:

```
Error:  Failures: 
Error:    AppReproducersTest.imageioAWTTest:427->imageioAWT:550 A different set of static libraries was expected. 
Expected: [libawt.a, libawt_headless.a, libfdlibm.a, libfontmanager.a, libjava.a, libjavajpeg.a, libjvm.a, liblcms.a, liblibchelper.a, libmanagement_ext.a, libnet.a, libnio.a, libzip.a]
Actual:   [libawt.a, libawt_headless.a, libfdlibm.a, libfontmanager.a, libjava.a, libjavajpeg.a, libjvm.a, liblcms.a, liblibchelper.a, libnet.a, libnio.a, libzip.a] ==> expected: <true> but was:
```
[See https://github.com/graalvm/mandrel/actions/runs/3148587052/jobs/5119491334](https://github.com/graalvm/mandrel/actions/runs/3148587052/jobs/5119491334#step:10:18565)

Related #109

This doesn't seem deterministic?

One run from https://github.com/graalvm/mandrel/actions/runs/3140760029/jobs/5102715980:

```
2022-09-28T06:13:07.3792360Z ========================================================================================================================
2022-09-28T06:13:07.3800679Z GraalVM Native Image: Generating 'target/imageio' (executable)...
2022-09-28T06:13:07.3801926Z ========================================================================================================================
2022-09-28T06:13:09.2820606Z Warning: Could not resolve com.orsoncharts.Chart3DHints for reflection configuration. Reason: java.lang.ClassNotFoundException: com.orsoncharts.Chart3DHints.
2022-09-28T06:13:10.6534221Z [1/7] Initializing...                                                                                    (6.1s @ 0.21GB)
2022-09-28T06:13:10.6538311Z  Version info: 'GraalVM 23.0.0-dev Java 17 CE'
2022-09-28T06:13:10.6541348Z  Java version info: '17.0.5+3-jvmci-22.3-b04'
2022-09-28T06:13:10.6548228Z  C compiler: gcc (linux, x86_64, 9.4.0)
2022-09-28T06:13:10.6549932Z  Garbage collector: Serial GC
2022-09-28T06:13:49.3124760Z [2/7] Performing analysis...  [******]                                                                  (38.4s @ 1.69GB)
2022-09-28T06:13:49.3163637Z    4,845 (78.09%) of  6,204 classes reachable
2022-09-28T06:13:49.3201174Z    8,176 (59.48%) of 13,745 fields reachable
2022-09-28T06:13:49.3402596Z   22,788 (49.61%) of 45,930 methods reachable
2022-09-28T06:13:49.3407932Z      179 classes,     6 fields, and   603 methods registered for reflection
2022-09-28T06:13:49.3413066Z      134 classes,   217 fields, and   119 methods registered for JNI access
2022-09-28T06:13:49.3420413Z        7 native libraries: dl, freetype, m, pthread, rt, stdc++, z
2022-09-28T06:13:54.3870904Z [3/7] Building universe...                                                                               (5.0s @ 1.10GB)
2022-09-28T06:13:58.9907957Z [4/7] Parsing methods...      [**]                                                                       (4.5s @ 0.92GB)
2022-09-28T06:14:01.6920062Z [5/7] Inlining methods...     [***]                                                                      (2.5s @ 1.71GB)
2022-09-28T06:14:35.5145106Z [6/7] Compiling methods...    [******]                                                                  (33.8s @ 1.25GB)
2022-09-28T06:14:40.0088767Z [7/7] Creating image...                                                                                  (2.5s @ 1.94GB)
2022-09-28T06:14:40.0097775Z   10.11MB (39.33%) for code area:    14,071 compilation units
2022-09-28T06:14:40.0103997Z   12.66MB (49.25%) for image heap:  152,370 objects and 13 resources
2022-09-28T06:14:40.0114611Z    2.94MB (11.42%) for other data
2022-09-28T06:14:40.0119166Z   25.70MB in total
2022-09-28T06:14:40.0123760Z ------------------------------------------------------------------------------------------------------------------------
2022-09-28T06:14:40.0134751Z Top 10 packages in code area:                               Top 10 object types in image heap:
2022-09-28T06:14:40.0162476Z  781.04KB java.util                                            2.12MB byte[] for code metadata
2022-09-28T06:14:40.0167907Z  697.39KB sun.font                                             1.47MB java.lang.String
2022-09-28T06:14:40.0173750Z  448.67KB com.sun.imageio.plugins.tiff                         1.41MB byte[] for embedded resources
2022-09-28T06:14:40.0179139Z  407.40KB com.sun.crypto.provider                              1.20MB byte[] for general heap data
2022-09-28T06:14:40.0189473Z  367.46KB java.lang                                            1.10MB java.lang.Class
2022-09-28T06:14:40.0194852Z  361.00KB java.awt.image                                     969.75KB byte[] for java.lang.String
2022-09-28T06:14:40.0200171Z  313.99KB com.sun.imageio.plugins.jpeg                       499.92KB java.util.HashMap$Node
2022-09-28T06:14:40.0205548Z  292.98KB sun.java2d.marlin                                  416.37KB com.oracle.svm.core.hub.DynamicHubCompanion
2022-09-28T06:14:40.0226641Z  279.65KB java.text                                          273.43KB java.lang.String[]
2022-09-28T06:14:40.0232410Z  271.42KB com.sun.imageio.plugins.png                        266.20KB java.util.concurrent.ConcurrentHashMap$Node
2022-09-28T06:14:40.0258248Z    5.88MB for 184 more packages                                2.33MB for 1005 more object types
2022-09-28T06:14:40.0318538Z ------------------------------------------------------------------------------------------------------------------------
2022-09-28T06:14:40.0339659Z                         2.1s (2.2% of total time) in 25 GCs | Peak RSS: 2.73GB | CPU load: 1.96
2022-09-28T06:14:40.0348631Z ------------------------------------------------------------------------------------------------------------------------
2022-09-28T06:14:40.0353148Z Produced artifacts:
2022-09-28T06:14:40.0406245Z  /home/runner/work/mandrel/mandrel/mandrel-integration-tests/apps/imageio/target/imageio (executable)
2022-09-28T06:14:40.0417745Z  /home/runner/work/mandrel/mandrel/mandrel-integration-tests/apps/imageio/target/imageio.build_artifacts.txt (txt)
2022-09-28T06:14:40.0421860Z ========================================================================================================================
2022-09-28T06:14:40.0429249Z Finished generating 'target/imageio' in 1m 35s.
2022-09-28T06:14:40.2157615Z 2022-09-28 06:14:40.215 INFO  [o.g.t.i.AppReproducersTest] (imageioAWT) Running...
2022-09-28T06:14:42.9575355Z [ERROR] Tests run: 10, Failures: 2, Errors: 0, Skipped: 3, Time elapsed: 529.469 s <<< FAILURE! - in org.graalvm.tests.integration.AppReproducersTest
2022-09-28T06:14:42.9579054Z [ERROR] resLocationsC{TestInfo}  Time elapsed: 62.761 s  <<< FAILURE!
2022-09-28T06:14:42.9631907Z org.opentest4j.AssertionFailedError: 
2022-09-28T06:14:42.9632473Z Something changed in how the resources are handled. Check comments on https://github.com/oracle/graal/issues/4326 and https://github.com/quarkusio/quarkus/pull/22403. See discrepancies: 
2022-09-28T06:14:42.9632917Z Expected: 3:  N/A, Actual: 3:  JDK
2022-09-28T06:14:42.9633168Z Expected: 14: N/A, Actual: 14: JDK
2022-09-28T06:14:42.9633400Z  ==> expected: <true> but was: <false>
2022-09-28T06:14:42.9633835Z 	at org.graalvm.tests.integration.AppReproducersTest.resLocations(AppReproducersTest.java:339)
2022-09-28T06:14:42.9634432Z 	at org.graalvm.tests.integration.AppReproducersTest.resLocationsC(AppReproducersTest.java:304)
2022-09-28T06:14:42.9634758Z 
2022-09-28T06:14:42.9634920Z [ERROR] imageioAWTTest{TestInfo}  Time elapsed: 107.302 s  <<< FAILURE!
2022-09-28T06:14:42.9635244Z org.opentest4j.AssertionFailedError: 
2022-09-28T06:14:42.9635557Z A different set of static libraries was expected. 
2022-09-28T06:14:42.9636041Z Expected: [libawt.a, libawt_headless.a, libfdlibm.a, libfontmanager.a, libjava.a, libjavajpeg.a, libjvm.a, liblcms.a, liblibchelper.a, libmanagement_ext.a, libnet.a, libnio.a, libzip.a]
2022-09-28T06:14:42.9636699Z Actual:   [libawt.a, libawt_headless.a, libfdlibm.a, libfontmanager.a, libjava.a, libjavajpeg.a, libjvm.a, liblcms.a, liblibchelper.a, libnet.a, libnio.a, libzip.a] ==> expected: <true> but was: <false>
```

FAIL.

Then compare to this one from https://github.com/graalvm/mandrel/actions/runs/3148587052/jobs/5119511668

```
2022-09-29T06:15:29.2464527Z ========================================================================================================================
2022-09-29T06:15:29.2474488Z GraalVM Native Image: Generating 'target/imageio' (executable)...
2022-09-29T06:15:29.2474855Z ========================================================================================================================
2022-09-29T06:15:31.2676054Z Warning: Could not resolve com.orsoncharts.Chart3DHints for reflection configuration. Reason: java.lang.ClassNotFoundException: com.orsoncharts.Chart3DHints.
2022-09-29T06:15:32.7237996Z [1/7] Initializing...                                                                                    (6.3s @ 0.21GB)
2022-09-29T06:15:32.7248769Z  Version info: 'GraalVM 23.0.0-dev Java 17 CE'
2022-09-29T06:15:32.7250776Z  Java version info: '17.0.5+3-jvmci-22.3-b04'
2022-09-29T06:15:32.7258970Z  C compiler: gcc (linux, x86_64, 9.4.0)
2022-09-29T06:15:32.7260137Z  Garbage collector: Serial GC
2022-09-29T06:16:12.8472103Z [2/7] Performing analysis...  [******]                                                                  (39.8s @ 1.65GB)
2022-09-29T06:16:12.8515387Z    4,845 (78.11%) of  6,203 classes reachable
2022-09-29T06:16:12.8567119Z    8,176 (59.48%) of 13,745 fields reachable
2022-09-29T06:16:12.8775528Z   22,788 (49.61%) of 45,935 methods reachable
2022-09-29T06:16:12.8781841Z      179 classes,     6 fields, and   603 methods registered for reflection
2022-09-29T06:16:12.8786853Z      134 classes,   217 fields, and   119 methods registered for JNI access
2022-09-29T06:16:12.8792228Z        7 native libraries: dl, freetype, m, pthread, rt, stdc++, z
2022-09-29T06:16:18.1041361Z [3/7] Building universe...                                                                               (5.2s @ 1.06GB)
2022-09-29T06:16:22.9586946Z [4/7] Parsing methods...      [**]                                                                       (4.8s @ 0.91GB)
2022-09-29T06:16:25.8947457Z [5/7] Inlining methods...     [***]                                                                      (2.8s @ 1.68GB)
2022-09-29T06:17:03.6325962Z [6/7] Compiling methods...    [******]                                                                  (37.7s @ 1.32GB)
2022-09-29T06:17:08.6878154Z [7/7] Creating image...                                                                                  (2.8s @ 2.02GB)
2022-09-29T06:17:08.6889607Z   10.11MB (39.32%) for code area:    14,070 compilation units
2022-09-29T06:17:08.6895727Z   12.66MB (49.25%) for image heap:  152,410 objects and 13 resources
2022-09-29T06:17:08.6908459Z    2.94MB (11.42%) for other data
2022-09-29T06:17:08.6913260Z   25.70MB in total
2022-09-29T06:17:08.6917837Z ------------------------------------------------------------------------------------------------------------------------
2022-09-29T06:17:08.6928526Z Top 10 packages in code area:                               Top 10 object types in image heap:
2022-09-29T06:17:08.6950987Z  780.95KB java.util                                            2.12MB byte[] for code metadata
2022-09-29T06:17:08.6956684Z  697.38KB sun.font                                             1.47MB java.lang.String
2022-09-29T06:17:08.6962122Z  448.67KB com.sun.imageio.plugins.tiff                         1.41MB byte[] for embedded resources
2022-09-29T06:17:08.6967477Z  407.37KB com.sun.crypto.provider                              1.20MB byte[] for general heap data
2022-09-29T06:17:08.6972969Z  367.82KB java.lang                                            1.10MB java.lang.Class
2022-09-29T06:17:08.6978705Z  361.00KB java.awt.image                                     969.78KB byte[] for java.lang.String
2022-09-29T06:17:08.6984040Z  314.27KB com.sun.imageio.plugins.jpeg                       500.39KB java.util.HashMap$Node
2022-09-29T06:17:08.6989431Z  292.98KB sun.java2d.marlin                                  416.37KB com.oracle.svm.core.hub.DynamicHubCompanion
2022-09-29T06:17:08.6994910Z  279.62KB java.text                                          273.43KB java.lang.String[]
2022-09-29T06:17:08.7058459Z  271.42KB com.sun.imageio.plugins.png                        266.30KB java.util.concurrent.ConcurrentHashMap$Node
2022-09-29T06:17:08.7214534Z    5.88MB for 184 more packages                                2.33MB for 1005 more object types
2022-09-29T06:17:08.7255766Z ------------------------------------------------------------------------------------------------------------------------
2022-09-29T06:17:08.7274534Z                         2.2s (2.1% of total time) in 25 GCs | Peak RSS: 2.74GB | CPU load: 1.96
2022-09-29T06:17:08.7280044Z ------------------------------------------------------------------------------------------------------------------------
2022-09-29T06:17:08.7284948Z Produced artifacts:
2022-09-29T06:17:08.7417618Z  /home/runner/work/mandrel/mandrel/mandrel-integration-tests/apps/imageio/target/imageio (executable)
2022-09-29T06:17:08.7424257Z  /home/runner/work/mandrel/mandrel/mandrel-integration-tests/apps/imageio/target/imageio.build_artifacts.txt (txt)
2022-09-29T06:17:08.7428664Z ========================================================================================================================
2022-09-29T06:17:08.7434639Z Finished generating 'target/imageio' in 1m 42s.
2022-09-29T06:17:08.9166943Z 2022-09-29 06:17:08.915 INFO  [o.g.t.i.AppReproducersTest] (imageioAWT) Running...
2022-09-29T06:17:11.5857826Z 2022-09-29 06:17:11.584 INFO  [o.g.t.i.u.Logs] (checkLog) build-and-run.log log for imageioAWTTest contains whitelisted error: `Warning: Could not resolve com.orsoncharts.Chart3DHints for reflection configuration. Reason: java.lang.ClassNotFoundException: com.orsoncharts.Chart3DHints.'
2022-09-29T06:17:11.6318382Z [WARNING] Tests run: 11, Failures: 0, Errors: 0, Skipped: 4, Time elapsed: 554.57 s - in org.graalvm.tests.integration.AppReproducersTest
```
PASS.

Not sure what's going on...

--- a/testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java
@@ -529,14 +529,7 @@ public class AppReproducersTest {
             // Test static libs in the executable
             final File executable = new File(appDir.getAbsolutePath() + File.separator + "target", "imageio");
             Set<String> expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a", "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a", "libnet.a", "libnio.a", "libzip.a");
-            if (UsedVersion.getVersion(inContainer).compareTo(Version.create(22, 3, 0)) >= 0) {
-                // libmanagement_ext.a added in 22.2+ with https://github.com/oracle/graal/commit/a0e6a3aeb8b63f6c06dc3554c342075534d90796
-                // but it wasn't deemed reachable until by the later commit https://github.com/oracle/graal/commit/2e3a0ae220f202be8f92f8738b9ba3aea57aea7d
-                // Therefore we expect it only for 22.3+
-                Set<String> modifiable = new HashSet<>(expected);
-                modifiable.add("libmanagement_ext.a");
-                expected = Collections.unmodifiableSet(modifiable);
-            } else if (UsedVersion.jdkFeature(inContainer) > 11 || (UsedVersion.jdkFeature(inContainer) == 11 && UsedVersion.jdkUpdate(inContainer) > 12)) {
+            if (UsedVersion.jdkFeature(inContainer) > 11 || (UsedVersion.jdkFeature(inContainer) == 11 && UsedVersion.jdkUpdate(inContainer) > 12)) {
                 // Harfbuzz removed: https://github.com/graalvm/mandrel/issues/286
                 // NO-OP
             } else {

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
@@ -101,9 +101,9 @@ public enum BuildAndRunCmds {
     }),
     IMAGEIO(new String[][]{
             new String[]{"mvn", "clean", "package"},
-            new String[]{"java", "-agentlib:native-image-agent=config-output-dir=src/main/resources/META-INF/native-image", "-jar", "target/imageio.jar"},
+            new String[]{"java", "-Djava.awt.headless=true", "-agentlib:native-image-agent=config-output-dir=src/main/resources/META-INF/native-image", "-jar", "target/imageio.jar"},
             new String[]{"jar", "uf", "target/imageio.jar", "-C", "src/main/resources/", "META-INF"},
-            new String[]{"native-image", "-H:IncludeResources=Grace_M._Hopper.jp2,MyFreeMono.ttf,MyFreeSerif.ttf", "--no-fallback", "-jar", "target/imageio.jar", "target/imageio"},
+            new String[]{"native-image", "-J-Djava.awt.headless=true", "-H:IncludeResources=Grace_M._Hopper.jp2,MyFreeMono.ttf,MyFreeSerif.ttf", "--no-fallback", "-jar", "target/imageio.jar", "target/imageio"},
             new String[]{IS_THIS_WINDOWS ? "target\\imageio.exe" : "./target/imageio"}
     }),
     IMAGEIO_BUILDER_IMAGE(new String[][]{
@@ -112,7 +112,7 @@ public enum BuildAndRunCmds {
             // TODO: Ad -u: Test access rights with -u on Windows, Docker Desktop Hyper-V backend vs. WSL2 backend.
             // Java from Builder image container is used for the sake of consistence.
             new String[]{CONTAINER_RUNTIME, "run", IS_THIS_WINDOWS ? "" : "-u", IS_THIS_WINDOWS ? "" : getUnixUIDGID(),
-                    "-t", "--entrypoint", "java", "-v", BASE_DIR + File.separator + "apps" + File.separator + "imageio:/project:z",
+                    "-t", "--entrypoint", "java", "-Djava.awt.headless=true", "-v", BASE_DIR + File.separator + "apps" + File.separator + "imageio:/project:z",
                     BUILDER_IMAGE,
                     "-agentlib:native-image-agent=config-output-dir=src/main/resources/META-INF/native-image", "-jar", "target/imageio.jar"},
             // Jar could be used locally, but we use the one from container too.
@@ -124,7 +124,7 @@ public enum BuildAndRunCmds {
             new String[]{CONTAINER_RUNTIME, "run", IS_THIS_WINDOWS ? "" : "-u", IS_THIS_WINDOWS ? "" : getUnixUIDGID(),
                     "-t", "-v", BASE_DIR + File.separator + "apps" + File.separator + "imageio:/project:z",
                     BUILDER_IMAGE,
-                    "-H:IncludeResources=Grace_M._Hopper.jp2,MyFreeMono.ttf,MyFreeSerif.ttf", "--no-fallback", "-jar", "target/imageio.jar", "target/imageio"},
+                    "-J-Djava.awt.headless=true", "-H:IncludeResources=Grace_M._Hopper.jp2,MyFreeMono.ttf,MyFreeSerif.ttf", "--no-fallback", "-jar", "target/imageio.jar", "target/imageio"},
             // We build a runtime image, ubi 8 minimal based, runtime dependencies installed
             new String[]{CONTAINER_RUNTIME, "build", "--network=host", "-t", ContainerNames.IMAGEIO_BUILDER_IMAGE.name, "."},
             // We have to run int he same env as we run the java part above, i.e. in the same container base.

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
@@ -112,8 +112,8 @@ public enum BuildAndRunCmds {
             // TODO: Ad -u: Test access rights with -u on Windows, Docker Desktop Hyper-V backend vs. WSL2 backend.
             // Java from Builder image container is used for the sake of consistence.
             new String[]{CONTAINER_RUNTIME, "run", IS_THIS_WINDOWS ? "" : "-u", IS_THIS_WINDOWS ? "" : getUnixUIDGID(),
-                    "-t", "--entrypoint", "java", "-Djava.awt.headless=true", "-v", BASE_DIR + File.separator + "apps" + File.separator + "imageio:/project:z",
-                    BUILDER_IMAGE,
+                    "-t", "--entrypoint", "java", "-v", BASE_DIR + File.separator + "apps" + File.separator + "imageio:/project:z",
+                    BUILDER_IMAGE, "-Djava.awt.headless=true",
                     "-agentlib:native-image-agent=config-output-dir=src/main/resources/META-INF/native-image", "-jar", "target/imageio.jar"},
             // Jar could be used locally, but we use the one from container too.
             new String[]{CONTAINER_RUNTIME, "run", IS_THIS_WINDOWS ? "" : "-u", IS_THIS_WINDOWS ? "" : getUnixUIDGID(),

